### PR TITLE
[LETS-680] Handle the no-ps-conn error case in the callers

### PR DESCRIPTION
**Pull request by wrlawodms**
http://jira.cubrid.org/browse/LETS-680

When no PS connection is available, `send_receive` fails, but it isn't necessary and is bothersome in some cases. In this PR, the error case is got out of `send_receive` and handled in the caller's own way. The ER_CONN_NO_PAGE_SERVER_AVAILABLE is added.

**Following the existing error handling like other errors (no exceptional handling in the callers):**

- _SEND_LOG_BOOT_INFO_FETCH_: logpb_fatal_error
- _GET_OLDEST_ACTIVE_MVCCID_: just print an error msg and retry later
- _SEND_STOP_LOG_PRIOR_DISPATCH_: ignore
- _GET_BOOT_INFO_: fails to boot.

**assert_release:**

- _SEND_LOG_PAGE_FETCH_: assert_release
- _SEND_DATA_PAGE_FETCH_: assert_release

### CODE REVIEW

File: src/server/tran_server.cpp [hunk 1]
```diff
@@ -145,7 +145,7 @@ tran_server::push_request (tran_to_page_request reqid, std::string &&payload)
   std::shared_lock<std::shared_mutex> s_lock (m_page_server_conn_vec_mtx);
   if (m_page_server_conn_vec.empty ())
     {
-      assert_release (false); // TODO some error-handling is needed such as shutdown
+      er_set (ER_ERROR_SEVERITY, ARG_FILE_LINE, ER_NO_PAGE_SERVER_CONNECTION, 0);
```
Comment (cristiarg) [inline src/server/tran_server.cpp]: The message for this error code reads:
```
At least one valid page server host is necessary, but no page server could be connected.
```
I think another one is needed.
Eg: `ER_NO_PAGE_SERVER_CONNECTION_REMAINING`.

File: src/transaction/log_page_buffer.c [hunk 2]
```diff
@@ -2115,6 +2115,10 @@ logpb_request_log_page_from_page_server (LOG_PAGEID log_pageid, LOG_PAGE * log_p
   // client side communication to page server error
   if (error_code != NO_ERROR)
     {
+      if (error_code == ER_NO_PAGE_SERVER_CONNECTION)
```
Comment (cristiarg) [inline src/transaction/log_page_buffer.c]: isn't it better to move this after the `if (perform_logging)` scope?
same above

**Review by joohok — APPROVED** [review]

File: src/server/tran_server.cpp (hunk 1 above)
Comment (wrlawodms) [reply src/server/tran_server.cpp]: OK.

File: src/transaction/log_page_buffer.c (hunk 2 above)
Comment (wrlawodms) [reply src/transaction/log_page_buffer.c]: Done

File: src/server/tran_server.cpp [hunk 3]
```diff
@@ -159,8 +159,8 @@ tran_server::send_receive (tran_to_page_request reqid, std::string &&payload_in,
   std::shared_lock<std::shared_mutex> s_lock (m_page_server_conn_vec_mtx);
   if (m_page_server_conn_vec.empty())
     {
-      assert_release (false); // TODO some error-handling is needed such as shutdown
-      return ER_FAILED; // All connections have been disconnected already
+      er_set (ER_ERROR_SEVERITY, ARG_FILE_LINE, ER_CONN_NO_PAGE_SERVER_AVAILABLE, 0);
+      return ER_NO_PAGE_SERVER_CONNECTION;
```
Comment (hornetmj) [inline src/server/tran_server.cpp]:
```suggestion
      return ER_CONN_NO_PAGE_SERVER_AVAILABLE;
```

**Review by hornetmj — APPROVED** [review]

File: src/server/tran_server.cpp (hunk 3 above)
Comment (wrlawodms) [reply src/server/tran_server.cpp]: ㅇ0ㅇ thank you.

File: msg/en_US.utf8/cubrid.msg [hunk 4]
```diff
@@ -1347,7 +1347,7 @@ Check the path of the key file (_keys) and make sure that it includes proper key
 1276 Connected page server cannot be reached anymore.
 1277 Atomic replication error: %s.
 1278 Decompression failed (%s).
-1279 Error reserved for scalability development.
+1279 No page server connection is available. At least one connection is necessary.
```
Comment (cristiarg) [inline msg/en_US.utf8/cubrid.msg]: Nit.
To me, it sounds better `required` or `needed`.

File: msg/en_US/cubrid.msg [hunk 5]
```diff
@@ -1347,7 +1347,7 @@ Check the path of the key file (_keys) and make sure that it includes proper key
 1276 Connected page server cannot be reached anymore.
 1277 Atomic replication error: %s.
 1278 Decompression failed (%s).
-1279 Error reserved for scalability development.
+1279 No page server connection is available. At least one connection is necessary.
```
Comment (cristiarg) [inline msg/en_US/cubrid.msg]: Shouldn't all the other language msg files be edited? That is how I know. But maybe you know better.

**Review by cristiarg — APPROVED** [review]

File: msg/en_US/cubrid.msg (hunk 5 above)
Comment (wrlawodms) [reply msg/en_US/cubrid.msg]: Yes, you're right. They'll be edited after the review too. I should've mentioned it.

**Pull request merged**